File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    private val matrixAsList = matrixAsString.lines().map { it.split(" ").filter(String::isNotEmpty).map(String::toInt) }

    fun column(colNr: Int): List<Int> = matrixAsList.map { it.elementAt(colNr - 1) }

    fun row(rowNr: Int): List<Int> = matrixAsList[rowNr - 1]

}

File: ruby/matrix/matrix.rb
class Matrix
  def initialize(matrix)
    @rows = matrix
            .split(/\n/)
            .map { |r| r.split(' ').map(&:to_i) }

    @columns = @rows.transpose
  end

  def row(row_number)
    @rows[row_number - 1]
  end

  def column(col_number)
    @columns[col_number - 1]
  end
end

